test/flows/student_finance_calculator_flow_test.rb

```diff
@@ -0,0 +1,519 @@
+          add_responses what_type_of_student_are_you?: full_time_student_type
+          max_for_full_time = SmartAnswer::Calculators::StudentFinanceCalculator::TUITION_FEE_MAXIMUM["full-time"]
+          assert_invalid_response (max_for_full_time + 1).to_s
+        end
+      end
+
+      %w[uk-part-time eu-part-time].each do |part_time_student_type|
+        should "be invalid for a fee greater than the maximum for a #{part_time_student_type} student" do
+          add_responses what_type_of_student_are_you?: part_time_student_type
+          max_for_part_time = SmartAnswer::Calculators::StudentFinanceCalculator::TUITION_FEE_MAXIMUM["part-time"]
+          assert_invalid_response (max_for_part_time + 1).to_s
+        end
+      end
+    end
+
+    context "next_node" do
+      %w[uk-full-time uk-part-time].each do |uk_student_type|
+        should "have a next node of where_will_you_live_while_studying? for a #{uk_student_type} student" do
+          add_responses what_type_of_student_are_you?: uk_student_type
+          assert_next_node :where_will_you_live_while_studying?, for_response: "5000"
+        end
+      end
+
+      %w[eu-full-time eu-part-time].each do |eu_student_type|
+        should "have a next node of outcome_eu_students for a #{eu_student_type} student" do
+          add_responses what_type_of_student_are_you?: eu_student_type
+          assert_next_node :outcome_eu_students, for_response: "5000"
+        end
+      end
+    end
+  end
+
+  context "question: where_will_you_live_while_studying?," do
+    setup do
+      testing_node :where_will_you_live_while_studying?
+      add_responses when_does_your_course_start?: "2020-2021",
+                    what_type_of_student_are_you?: "uk-full-time",
+                    how_much_are_your_tuition_fees_per_year?: "9250"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of whats_your_household_income? for any response" do
+        assert_next_node :whats_your_household_income?, for_response: "at-home"
+      end
+    end
+  end
+
+  context "question: whats_your_household_income?," do
+    setup do
+      testing_node :whats_your_household_income?
+      add_responses when_does_your_course_start?: "2020-2021",
+                    what_type_of_student_are_you?: "uk-full-time",
+                    how_much_are_your_tuition_fees_per_year?: "9250",
+                    where_will_you_live_while_studying?: "at-home"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node do_any_of_the_following_apply_uk_full_time_students_only? for any response" do
+        assert_next_node :do_any_of_the_following_apply_uk_full_time_students_only?, for_response: "50,000"
+      end
+
+      should "have a next node of how_many_credits_will_you_study?" do
```

Comment:
I'd add "for part time students" to this line and "for full time students" on 124 just to make it a bit clearer what's being tested